components/CollectionCard.js

```diff
@@ -159,6 +157,16 @@ export default function CollectionCard({
     setShowConfirmation(!showConfirmation);
   }
 
+  function handleSubmit(id, data) {
+    handleUpdateCollection(id, data);
```

Comment:
You actually don't need to pass the `id` separately here as the `id` is already included in `data`.